sdcm/sct_config.py

```diff
@@ -1007,12 +1007,9 @@ class SCTConfiguration(dict):
         dict(name="cluster_health_check", env="SCT_CLUSTER_HEALTH_CHECK", type=boolean,
              help="When true, start cluster health checker for all nodes"),
 
-        dict(name="validate_partitions", env="SCT_VALIDATE_PARTITIONS", type=boolean,
-             help="when true, log of the partitions before and after the nemesis run is compacted"),
-        dict(name="table_name", env="SCT_TABLE_NAME", type=str,
-             help="table name to check for the validate_partitions check"),
-        dict(name="primary_key_column", env="SCT_PRIMARY_KEY_COLUMN", type=str,
-             help="primary key of the table to check for the validate_partitions check"),
+        dict(name="data_validation", env="SCT_DATA_VALIDATION", type=str,
+             help="A group of sub-parameters: validate_partitions, table_name, primary_key_column,"
```

Comment:
`A group of sub-parameters` and the values for them? are they "mandatory" in all large partition test? should these options be in a strict group (i think better not)?

Reply:
please see its description in configuration_options.md like:
``` | **<a href="#user-content-data_validation" name="data_validation">data_validation</a>** | A group of sub-parameters (see below): validate_partitions, table_name, primary_key_column, partition_range_with_data_validation, max_partitions_in_test_table | N/A | SCT_DATA_VALIDATION
```

Reply:
what's writen in the docs, is coming from this, not sure what's the argument, and how it's answering @fgelcer question

Reply:
@fruch , not in this case, since it is a kind of a new format of "sub-parameters". So they are only present in `configuration_options.md`  like:
```
| **<a href="#user-content-data_validation" name="data_validation">data_validation</a>** | A group of sub-parameters (see below): validate_partitions, table_name, primary_key_column, partition_range_with_data_validation, max_partitions_in_test_table | N/A | SCT_DATA_VALIDATION
| **<a href="#user-content-validate_partitions" name="validate_partitions"></a>** | Sub-parameter: validate_partitions - when true, log of the partitions before and after the nemesis run is compacted | N/A | N/A
| **<a href="#user-content-table_name" name="table_name"></a>** | Sub-parameter: table_name - table name to check for the validate_partitions check | N/A | N/A
| **<a href="#user-content-primary_key_column" name="primary_key_column"></a>** | Sub-parameter: primary_key_column - primary key of the table to check for the validate_partitions check| N/A | N/A
| **<a href="#user-content-partition_range_with_data_validation" name="partition_range_with_data_validation"></a>** | Sub parameter: partition_range_with_data_validation - Relevant for scylla-bench. Hold range (min - max) of PKs values for partitions that data was<br>written with validate data and will be validate during the read.<br>Example: 0-250.<br>Optional parameter for DeleteByPartitionsMonkey and DeleteByRowsRangeMonkey| N/A | N/A
| **<a href="#user-content-max_partitions_in_test_table" name="max_partitions_in_test_table"></a>** | Sub parameter: max_partitions_in_test_table - Relevant for scylla-bench. MAX partition keys (partition-count) in the scylla_bench.test table.<br>Mandatory parameter for DeleteByPartitionsMonkey and DeleteByRowsRangeMonkey| N/A | N/A
```

Reply:
So this is not acceptable, that file is autogenerated, next time someone add new configuration, your docs would be gone.

If you want to document this internal structure, it should all be included in the parameter mentioned in sct_config.py

If you want to define them all, the need code to provide it should be supplied

Reply:
@fruch , fixed by moving this documentation to `sct_config.py` like:
```
-             help="A group of sub-parameters: validate_partitions, table_name, primary_key_column,"
-                  " partition_range_with_data_validation, max_partitions_in_test_table"),
+             help="""A group of sub-parameters: validate_partitions, table_name, primary_key_column,
+                   partition_range_with_data_validation, max_partitions_in_test_table.
+                   1. validate_partitions - when true, validating the same number of rows-per-partition before/after a Nemesis.
+                   2. table_name - table name to check for the validate_partitions check.
+                   3. primary_key_column - primary key of the table to check for the validate_partitions check
+                   4. partition_range_with_data_validation - Relevant for scylla-bench. A range (min - max) of PK values
+                       for partitions to be validated by reads and not to be deleted during test. Example: 0-250.
+                   5. max_partitions_in_test_table - Relevant for scylla-bench. Max partition keys (partition-count)
+                       in the scylla_bench.test table.
+                  """),






-| **<a href="#user-content-validate_partitions" name="validate_partitions"></a>** | Sub-parameter: validate_partitions - when true, log of the partitions before and after the nemesis run is compacted | N/A | N/A
-| **<a href="#user-content-table_name" name="table_name"></a>** | Sub-parameter: table_name - table name to check for the validate_partitions check | N/A | N/A
-| **<a href="#user-content-primary_key_column" name="primary_key_column"></a>** | Sub-parameter: primary_key_column - primary key of the table to check for the validate_partitions check| N/A | N/A
-| **<a href="#user-content-partition_range_with_data_validation" name="partition_range_with_data_validation"></a>** | Sub parameter: partition_range_with_data_validation - Relevant for scylla-bench. Hold range (min - max) of PKs values for partitions that data was<br>written with validate data and will be validate during the read.<br>Example: 0-250.<br>Optional parameter for DeleteByPartitionsMonkey and DeleteByRowsRangeMonkey| N/A | N/A
-| **<a href="#user-content-max_partitions_in_test_table" name="max_partitions_in_test_table"></a>** | Sub parameter: max_partitions_in_test_table - Relevant for scylla-bench. MAX partition keys (partition-count) in the scylla_bench.test table.<br>Mandatory parameter for DeleteByPartitionsMonkey and DeleteByRowsRangeMonkey| N/A | N/A
```

Reply:
ping-after-2-weeks  @fruch  , @fgelcer

Reply:
> ping-after-2-weeks @fruch , @fgelcer

ping-after-2-weeks @fruch , @fgelcer